File: ios-native/NeobankNative/Sources/Features/Wallet/WalletModels.swift
import Foundation

/// Amounts are decimal strings straight from the ledger (never `Double`, to
/// avoid floating-point drift) — formatted for display, never parsed and
/// re-computed on-device.
struct WalletBalance: Decodable, Sendable {
    let walletId: String
    let currency: String
    let balance: String
    let availableBalance: String
    let encumberedBalance: String?
}

struct WalletTransaction: Decodable, Sendable, Identifiable {
    let id: String
    let type: String
    let amount: String
    let currency: String
    let direction: String
    let status: String
    let createdAt: Date
    let counterparty: String?
    let memo: String?
    let referenceId: String?

    var isCredit: Bool { direction == "credit" }
}

struct WalletTransactionPage: Decodable, Sendable {
    let transactions: [WalletTransaction]
    let nextCursor: String?
}

File: android-native/app/src/main/kotlin/dev/horobets/neobanknative/features/wallet/WalletModels.kt
package dev.horobets.neobanknative.features.wallet

import kotlinx.serialization.Serializable

/**
 * Amounts are decimal strings straight from the ledger (never [Double], to
 * avoid floating-point drift) — formatted for display, never parsed and
 * re-computed on-device.
 */
@Serializable
data class WalletBalance(
    val walletId: String,
    val currency: String,
    val balance: String,
    val availableBalance: String,
    val encumberedBalance: String? = null,
)

@Serializable
data class WalletTransaction(
    val id: String,
    val type: String,
    val amount: String,
    val currency: String,
    val direction: String,
    val status: String,
    val createdAt: String,
    val counterparty: String? = null,
    val memo: String? = null,
    val referenceId: String? = null,
) {
    val isCredit: Boolean get() = direction == "credit"
}

@Serializable
data class WalletTransactionPage(
    val transactions: List<WalletTransaction>,
    val nextCursor: String? = null,
)
